src/trapfunc.cpp

```diff
@@ -605,14 +607,22 @@ void trapfunc::goo( Creature *c, const tripoint &p )
 
 void trapfunc::dissector( Creature *c, const tripoint &p )
 {
-    //~ the sound of a dissector dissecting
-    sounds::sound( p, 10, _( "BRZZZAP!" ) );
     if( c != nullptr ) {
+        monster *z = dynamic_cast<monster *>( c );
+        if( z != nullptr && z->type->in_species( ROBOT ) ){
+            //The monster is a robot. So the dissector should not try to dissect the monsters flesh.
+            sounds::sound( p, 4, _( "BEEPBOOP! Please remove non-organic object. " ) ); //Dissector error sound.
+            c->add_msg_player_or_npc( m_bad, _( "The dissector lights up, and shuts down." ),
+                                  _( "The dissector lights up, and shuts down" ) );
```

Comment:
Yes, the if statement above makes it impossible for the C to be a player or npc, but the add_msg_player_or_npc function also works on monsters. So it works.